frontend/packages/kubevirt-plugin/src/components/vm-snapshots/vm-snapshot-details.tsx

```diff
@@ -58,7 +59,7 @@ const SnapshotDetails: React.FC<DetailsProps> = ({ obj, customData: { restores }
                 }
               >
                 {obj?.metadata?.annotations?.description || (
-                  <div className="text-muted">{NOT_AVAILABLE_MESSAGE}</div>
+                  <div className="text-muted">{t('kubevirt-plugin~Not available')}</div>
```

Comment:
would it be possible to extract the keys into separate file ?
For example - we would create `translation-keys.ts`
```js
export const NOT_AVAILABLE = 'kubevirt-plugin~Not available'
```
and then use it `t(NOT_AVAILABLE)` ?

Reply:
> use it t(NOT_AVAILABLE) ?

No we can't, AFAIU the translation guidelines is to have clear texts in the `t` functions. so we must convert all our text constants to texts as part of the i18n effort.

To enforce this rule, the `yarn i18n` script will not translate this strings. 

@pcbailey am I currect ?

Reply:
I asked about doing this. We could do it, but only with a hack. In general, it's being discouraged, but not forbidden. Below is Sam's response when I asked about our constants files. I'm open to the conversation if we want to do it a different way. I don't particularly like the inline approach. 

> You’ll have to refactor that. If you use the global i18next instance there, the strings won’t update on language change until after a page refresh. You could change the constants to refer to a translation key and use it like `t(CUSTOM_FLAVOR_LABEL_KEY)`. If you do that, you’ll need a comment in the code like `// t('kubevirt-plugin~Custom')` so the i18next parser knows about the string.
> 
> If you’re only using them one or two places, I’d just inline the keys in the component. Or you could refactor the code so that commonly used strings are in a reusable component.

Reply:
I like the inline approach, 
a - it's more clear when reading the file what is the tring shown.
b - the strings are actually keys, for the constant strings stored in the i18n `.json` file, so keeping another constant file for strings will be actually a duplication.

@pcbailey @rawagner can we keep it inline ?

Reply:
The primary benefit to using constants file(s) like we currently have, is that you can change the string in one place and it updates for everyone. The i18n file doesn't work like that. Since the string is the key, it will still need to be changed everywhere it's used. I'm ok with using the inline approach if @rawagner is, though.

But, I did think of another option that may work. What if we convert the constants to functions that take the `t()` function, like this:

`const NOT_AVAILABLE = (t: TFunction) => t('kubevirt-plugin~Not available');`

Reply:
I prefer the inline approach,
a - the dictionary file is in our code base (under `kubevirt-plugin`) and we can see every new string added.
so if we forget to change a string somware, we will imidiatly see it as a new string that appear.

for example I had two strings:
`"{{label}} Description"` and `"{{name}} Descriotion"`

I Immediately saw that I have two similar strings in the `.json` file and converged them to be one string.

b - the hopes that we need to jump through in order to have a common place for strings ( e.g. moving the `t` function around) and having two files with strings are "not nice" IMHO.

Reply:
looking at guidelines I think having file with keys should work

translation-keys.ts
```js
export const NOT_AVAILABLE = 'kubevirt-plugin~Not available'
```

foo-component.tsx
```js
<div className="text-muted">{t(NOT_AVAILABLE)}</div>
```



what does not work is if we want to convert our existing constants to use `t` directly like

```
export const NOT_AVAILABLE = t('kubevirt-plugin~Not available');
```
foo-component.tsx (does not require any change) - this wont update if we change language
```js
<div className="text-muted">{NOT_AVAILABLE}</div>
```

But I didnt try it, so I may be wrong :) 

I just want to make sure we consider extracting keys and using constants. I think there's less chance to have typos but on the other hand looking at the core console we wont be aligned with them as they just use strings (and not being aligned with core is IMO is a big disadvantage and because of this I actually prefer strings).

Reply:
It sounds like everyone is ok with strings, so strings it is. =)